gemma-core/src/main/java/ubic/gemma/persistence/service/expression/experiment/ExpressionExperimentServiceImpl.java

```diff
@@ -1153,4 +1146,78 @@ public Boolean isSuitableForDEA( ExpressionExperiment ee ) {
         return this.expressionExperimentDao.getExperimentsLackingPublications();
     }
 
+    /**
+     * Checks for special properties that are allowed to be referenced on certain objects. E.g. characteristics on EEs.
+     * {@inheritDoc}
+     */
+    @Override
+    public ObjectFilter getObjectFilter( String propertyName, ObjectFilter.Operator operator, String requiredValue ) {
+        try {
+            AliasPropertyNameType ap = getAliasForProperty( propertyName );
+            return ObjectFilter.parseObjectFilter( ap.getObjectAlias(), ap.propertyName, ap.propertyType, operator, requiredValue );
+        } catch ( NoSuchFieldException e ) {
+            throw new IllegalArgumentException( e );
+        }
+    }
+
+    @Override
+    public ObjectFilter getObjectFilter( String propertyName, ObjectFilter.Operator operator, Collection<String> requiredValues ) {
+        try {
+            AliasPropertyNameType ap = getAliasForProperty( propertyName );
+            return ObjectFilter.parseObjectFilter( ap.getObjectAlias(), ap.propertyName, ap.propertyType, operator, requiredValues );
+        } catch ( NoSuchFieldException e ) {
+            throw new IllegalArgumentException( e );
+        }
+    }
+
+    @Override
+    public Sort getSort( String propertyName, Sort.Direction direction ) {
+        try {
+            AliasPropertyNameType ap = getAliasForProperty( propertyName );
+            return Sort.by( ap.getObjectAlias(), ap.getPropertyName(), direction );
+        } catch ( NoSuchFieldException e ) {
+            throw new IllegalArgumentException( e );
+        }
+    }
+
+    private AliasPropertyNameType getAliasForProperty( String propertyName ) throws NoSuchFieldException {
```

Comment:
I regrouped all the aliases we use in the frontend/RESTful API here so that they apply to both ObjectFilter and Sort.